src/codelabs/null-safety.md

```diff
@@ -0,0 +1,321 @@
+---
+title: Null safety codelab
+description: Learn about and practice writing null safe code in DartPad!
+js: [{url: 'https://dartpad.dev/inject_embed.dart.js', defer: true}]
+---
+{% assign useIframe = false -%}
+<?code-excerpt replace="/ *\/\/\s+ignore_for_file:[^\n]+\n//g; /(^|\n)*\/\/\s+ignore:[^\n]+\n/$1/g; /(\n[^\n]+) *\/\/\s+ignore:[^\n]+\n/$1\n/g"?>
+<?code-excerpt plaster="none"?>
+
+This codelab teaches you about Dart’s null-safe type system,
```

Comment:
s/Dart's null-safety type system/Dart's sound null safety feature/

(The content here is beyond the type system, and also talks about the tooling that supports the type system)